packages/ra-core/src/core/useGetRecordRepresentation.ts

```diff
@@ -31,6 +31,18 @@ export const useGetRecordRepresentation = (
             if (React.isValidElement(recordRepresentation)) {
                 return recordRepresentation;
             }
+            if ('name' in record) {
+                return record.name;
+            }
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏Instead of checking of the field exists in the record, you should check that it is not null nor undefined nor an empty string. 

```suggestion
            if (record?.name != null && record?.name !=== '') {
                return record.name;
            }
```

Same applies below.